### PR TITLE
Prevent on-death-action reloads from activating specials

### PR DESCRIPTION
**Pull request by kitchen-ace**
Co-Authored-By: JNechaevsky <julia.nechaevskaya@live.com>

### CODE REVIEW

**Comment by kitchen-ace** [comment]
not sure i did the co-author thing right, is that the right email @JNechaevsky ?

File: src/g_game.c [hunk 1]
```diff
@@ -2399,6 +2400,13 @@ static void G_DoLoadGame(void)
   // draw the pattern into the back screen
   R_FillBackScreen();
 
+  // [JN] If "On death action" is set to "last save",
+  // then prevent holded "use" button to work for next few tics.
+  // This fixes imidiate pressing on wall upon reloading
+  // a save game, if "use" button is kept pressed.
+  if (death_use_action == 1)
+      players[consoleplayer].usedown = true;
```
Comment (fabiangreffrath) [inline src/g_game.c:2408]: Setting `usedown` to `true` prevents triggering use actions?

**Comment by JNechaevsky** [comment]
It is, but something went wrong. 

I'll explain what's happening. The scenario is very simple, and it happens almost same in both Crispy and Woof implementations:
* Make a save near the exit switch.
* Go to next level.
* Kill player by something handy like Rocket Launcher.
* Press `use`. If key is holded for more than one tic (presumably), exit switch will be hilted immediately upload save loading.

That's it. To don't trigger exit switch, `use` button must be hit as quick as possible. Unfortunately, I wasn't able to find any other solution, but this seems to do the trick - it gives few tics delay if `use` is still pressed, but! Once `use` is released and pressed again, it will work fine. By "trick" I mean tricking [this condition](https://github.com/fabiangreffrath/woof/blob/master/src/p_user.c#L525).

**Comment by kitchen-ace** [comment]
>P.S. Important! It must be guarded by "single player" condition here.

I thought loading was only possible in single-player games in Woof:
https://github.com/fabiangreffrath/woof/blob/851bfcaefa9be68943347ff388795d36ad36ae5a/src/g_game.c#L2227C2-L2236C4
is that incorrect?

**Comment by rfomin** [comment]
> I thought loading was only possible in single-player games in Woof

It is disabled for loading from the menu (notable example is DEMO1 in AV.WAD - it's a multiplayer deathmatch, so in vanilla Doom loading from the menu won't work if the demo reel starts). It is possible to load a multiplayer save from the command line.

**Comment by rfomin** [comment]
How about this solution:
```diff
diff --git a/src/p_user.c b/src/p_user.c
index 0993f65..87525c6 100644
--- a/src/p_user.c
+++ b/src/p_user.c
@@ -21,6 +21,7 @@
 
 #include "doomstat.h"
 #include "d_event.h"
+#include "m_input.h"
 #include "r_main.h"
 #include "p_map.h"
 #include "p_spec.h"
@@ -340,6 +341,7 @@ void P_DeathThink (player_t* player)
           char *file = G_SaveGameName(savegameslot);
           G_LoadGame(file, savegameslot, false);
           free(file);
+          M_InputGameDeactivate(input_use);
         }
         else
           player->playerstate = PST_REBORN;
```

**Comment by kitchen-ace** [comment]
>How about this solution:

Hm, that looks like a much cleaner implementation. Does it need any special single-player guarding?

What do you think of that for inter-ports as well @JNechaevsky ?

**Comment by rfomin** [comment]
> Does it need any special single-player guarding?

No, it's already guarded at this point.

> What do you think of that for inter-ports 

Inter-ports probably don't have `M_InputGameDeactivate` function. It's simple, just disable the "use" key in the `gamekeydown` array.

**Comment by JNechaevsky** [comment]
Just checked @rfomin's suggestion - it works perfectly and acts same way as in initial implementation. Better use it, as it's native to Woof key handling and does not require extra `singleplayer` condition check.

**Comment by kitchen-ace** [comment]
@rfomin I'll just close this PR, you can make the change yourself or wait a day or two and I'll submit another one.

**Comment by JNechaevsky** [comment]
I'll take care of it. 🙂 What really make me wonder is why no one was complaining about it before. Either everyone is too familiar with `F3`/`Enter` key pressing, or it is just me who have a habit to make a save right before exit switch. Anyways, it's very useful feature, even in Heretic and notably in Hexen.

**Pull request closed without merging**